probe-rs/src/debug/unit_info.rs

```diff
@@ -119,63 +103,16 @@ impl UnitInfo {
             return Ok(vec![]);
         };
 
-        let mut current_depth = 0;
-        let mut abort_depth = 0;
         let mut functions = Vec::new();
 
-        while let Ok(Some((depth, current))) = cursor.next_dfs() {
-            current_depth += depth;
-            if current_depth < abort_depth {
-                break;
-            }
-
-            // Skip anything that is not an inlined subroutine.
-            if current.tag() != gimli::DW_TAG_inlined_subroutine {
+        while let Ok(Some((_depth, current))) = cursor.next_dfs() {
+            // We only want children of the DIE at the given offset (the non-inlined function DIE)
+            if current.offset() == offset {
```

Comment:
This check is not clear to me given the comment above:
`We only want children of the DIE at the given offset` - is the stress on `at the given offset` (i.e. is the `==` is wrong?) or on `children` (i.e. we expect the children to have a different offset)?

This question only exists because of my lack of domain knowledge, and I would appreciate some clarification.

Reply:
I will add some more comments ... it is justified. On reflection, I'm also going to restore the perf optimization of checking depth. I will ping you for a final review :)

Reply:
@bugadani Let me know if this helps to clarify the logic.